.github/workflows/code_quality.yml

```diff
@@ -4,10 +4,10 @@ on: [pull_request]
 
 jobs:
   code-quality:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
-          python-version: 3.9
+          python-version: 3.x
```

Comment:
Replace the python-versioning with a matrix config to ensure code quality across the earliest supported version (`3.9`) and the latest python version (`3.11`). E.g.
```yml
    strategy:
      matrix:
        python-version: ["3.9", "3.11"]

    steps:
      - uses: actions/checkout@v3
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v4
        with:
          python-version: ${{ matrix.python-version }}
```

Reply:
The deprecated pre-commit action cannot handle multiple Python versions when the Python version is set in the pre-commit config.
> RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.9'

Reply:
Ah, that's annoying. I wonder if updating the pre-commit rev would fix this?